data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -617,9 +614,28 @@ public void requireStoreAvailable() {
     return getValidatorSelector(state, validators)
         .filter(getStatusPredicate(state, statusFilter))
         .mapToObj(index -> ValidatorResponse.fromState(state, index))
```

Comment:
Maybe rather than having a `ValidatorResponse.EMPTY` `ValidatorResponse.fromState` could return an `Optional<ValidatorResponse>` then this could just flat map?

Reply:
it wanted `SafeFuture<Optional<Optional<ValidatorResponse>>`...

Reply:
That sounds like you need a `flatMap` somewhere instead of just `map`.